### PR TITLE
initial RPM packaging

### PR DESCRIPTION
**Pull request by sosiouxme**
I cargo-culted stuff just to get this working. Is there a more simplified template to follow?

### CODE REVIEW

**Comment by sosiouxme** [comment]
I got it to build in brew.

**Comment by sosiouxme** [comment]
@smarterclayton is this ok or would you like further refinement?

Need a brew built RPM for OSBS to use.

**Comment by smarterclayton** [comment]
If all you need is the spec file, I'd recommend just going the simple route
and strip this down to just a spec file, and just run go install in it.
Nothing about this tool really requires the extra ceremony.

**Comment by smarterclayton** [comment]
You shouldn't even need any of the cross arch stuff except to run GOARCH=X during your `go install`.  Also we don't want tito.

**Comment by sosiouxme** [comment]
> If all you need is the spec file, I'd recommend just going the simple route
> and strip this down to just a spec file, and just run go install in it.
>

I tried that, and it was just not obvious how to make it work. I probably
only needed some env vars set but well, packaging not my strength and
everyone else is busy.

We do want builds in multiple arches, but I suppose that would be handled
by running builds on multiple architectures...

I literally just copied code from
https://github.com/openshift/image-registry/ -- no one would point me at
anything simpler.

I can poke at it some more though to try to simplify.

**Comment by smarterclayton** [comment]
I would remove:

* hack/
* .tito/

Comments inline for spec

File: imagebuilder.spec [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+%prep
+%if 0%{do_prep}
+%setup -q
+%endif
+
+%build
+%if 0%{do_build}
+%if 0%{make_redistributable}
+# Create Binaries for all internally defined arches
+%{os_git_vars} hack/build-cross.sh
```
Comment (smarterclayton) [inline imagebuilder.spec]: Remove all this (and any os_git_vars reference), and remove the make_redistributable check.

In the if arch sections, replace the entire line with `GOOS=<value_of_first_part> GOARCH=<value_of_second_part> go install ./cmd/imagebuilder`

Then in the install step you'll need to move `GOPATH/bin/imagebuilder` into the right location.

File: imagebuilder.spec [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+%ifarch ppc64le
+  BUILD_PLATFORM="linux/ppc64le"
+%endif
+%ifarch %{arm} aarch64
+  BUILD_PLATFORM="linux/arm64"
+%endif
+%ifarch s390x
+  BUILD_PLATFORM="linux/s390x"
+%endif
+OS_ONLY_BUILD_PLATFORMS="${BUILD_PLATFORM}" %{os_git_vars} hack/build-cross.sh
```
Comment (smarterclayton) [inline imagebuilder.spec]: This would just be `go install ./cmd/imagebuilder`

File: imagebuilder.spec [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+%install
+
+PLATFORM="$(go env GOHOSTOS)/$(go env GOHOSTARCH)"
+install -d %{buildroot}%{_bindir}
+
+# Install linux components
+for bin in imagebuilder
+do
+  echo "+++ INSTALLING ${bin}"
+  install -p -m 755 _output/local/bin/${PLATFORM}/${bin} %{buildroot}%{_bindir}/${bin}
```
Comment (smarterclayton) [inline imagebuilder.spec]: This is the GOPATH/bin dir.

**Comment by sosiouxme** [comment]
How about this then? Just the spec file.

The trick was just to get everything in GOPATH.

**Comment by sosiouxme** [comment]
@smarterclayton poke about this

**Comment by smarterclayton** [comment]
Good enough for government work

**Pull request merged**